### PR TITLE
Buffer.alloc / Buffer.from

### PR DESCRIPTION
**Pull request by fanatid**
Issue: https://github.com/dominictarr/varstruct/issues/12

@dcousens please check when you will have free time

### CODE REVIEW

**Comment by dcousens** [comment]
Why movement to `lib/`?
I always found this confusing,  and often only to be seen in libraries that are using a transpiler for distribution...

File: lib/util.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 'use strict'
 
-// should be faster + work with undefined values
-exports.reduce = function (items, iter, acc) {
```
Comment (dcousens) [inline lib/util.js]: breaking change?

File: lib/varbuffer.js [hunk 2]
```diff
@@ -27,9 +27,9 @@ module.exports = function (lengthType) {
       offset += lengthType.decode.bytes
       if (offset + blength > end) throw new RangeError('not enough data for decode')
       decode.bytes = lengthType.decode.bytes + blength
-      return new Buffer(buffer.slice(offset, offset + blength))
+      return Buffer.from(buffer.slice(offset, offset + blength))
     },
-    encodingLength: function (value) {
+    encodingLength: function encodingLengthc (value) {
```
Comment (dcousens) [inline lib/varbuffer.js]: `encodingLengthc`?  Why the `c`

**Comment by dcousens** [comment]
ACK all but https://github.com/dominictarr/varstruct/pull/13/commits/87f6e3deea2396fb6edc57ff846eac44a42c1aba,  and nit on https://github.com/dominictarr/varstruct/pull/13/commits/3d5ee1bcdd7984f5bc88d31d3e7b41ff1b877ce9

File: lib/util.js (hunk 1 above)
Comment (fanatid) [inline lib/util.js]: it is internal function
but we need major bump anyway, right?

Comment (dcousens) [inline lib/util.js]: True

**Comment by fanatid** [comment]
> Why movement to `lib/`?
I always found this confusing,  and often only to be seen in libraries that are using a transpiler for distribution...

What you prefer? `src`? What about libraries where we have c++ bindings and js?

**Comment by dcousens** [comment]
@fanatid I actually just prefer the flat layout where possible.
In this case I suppose `src/` would be my preferred... but now i'm not sure either way.
Does `standard` define anything here? Haha

**Comment by dcousens** [comment]
ACK on `lib/`

**Comment by fanatid** [comment]
Published as `6.0.0`

**Pull request merged**